### PR TITLE
use reshape when possible in broadcasting

### PR DESCRIPTION
**Pull request by eellison**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #53327 [WIP] torch.utils.freeze
* #53908 [JIT] Add Reinplacing to MKLDNN Subgraphs
* **#53326 use reshape when possible in broadcasting**
* #52614 Avoid dispatch overhead in call to mkldnn convolution

Differential Revision: [D26897275](https://our.internmc.facebook.com/intern/diff/D26897275)

### CODE REVIEW

File: torch/csrc/jit/passes/frozen_ops_to_mkldnn.cpp [hunk 1]
```diff
@@ -49,13 +49,22 @@ Operation BroadOp(const Node* node) {
       push(stack, a, b);
     } else {
       auto out_size = at::infer_size(a_size, b_size);
+      size_t out_numel = out_size[0];
```
Comment (Krovatkin) [inline torch/csrc/jit/passes/frozen_ops_to_mkldnn.cpp:52]: I don't if we need to worry about 0-dim case here?

**Review by Krovatkin — APPROVED** [review]
:shipit:

File: torch/csrc/jit/passes/frozen_ops_to_mkldnn.cpp (hunk 1 above)
Comment (eellison) [reply torch/csrc/jit/passes/frozen_ops_to_mkldnn.cpp:52]: no, we dont fuse 0-dim tensors, we handle tensors with a dimension value of 0
